File: ADT/stack-queue-pblm/946.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        st = []
        n = len(pushed)
        i = 0
        j = 0
        while i < n and j < n:
            st.append(pushed[i])
            while st and j < n and st[-1] == popped[j]:
                st.pop()
                j += 1
            i += 1
        return not st

File: ADT/stack-queue-pblm/946.js
/**
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function (pushed, popped) {
    const st = []
    const n = pushed.length
    let i = 0
    let j = 0
    while (i < n && j < n) {
        st.push(pushed[i])
        while (st.length > 0 && j < n && st[st.length - 1] === popped[j]) {
            st.pop()
            j++
        }
        i++
    }
    return st.length === 0
};